internal/ctl/connectors/add.go

```diff
@@ -44,16 +46,25 @@ func doAddConnectors(_ *cobra.Command, params *addConnectorsCmdParams) {
 	clusterLogger.WithField("config", config).Trace("add connectors configuration")
 
 	var source manager.ConnectorSource
-	if params.Files != nil {
-		source = filesSource(&params.Files)
-	}
-	if params.Directory != "" {
-		source = directorySource(&params.Directory)
+
+	switch {
```

Comment:
Can we maybe avoid this code that is similar to the snippet below?

Reply:
:shrug I tired fixing this but the two instances are subtely different. It can be done but will need changing a few things beyond the scope of this change